### DISCUSSION TITLE
How to use a variable with "Action"

### DISCUSSION THREAD

**Discussion by lukasz2161** [hasura/graphql-engine#8765] [Question]
![Screenshot (81)](https://user-images.githubusercontent.com/65758825/183476082-eae68229-7398-4829-b9e5-10afa30e7db3.png)
Hi, How can I use the retrieved data from "Action" in other query?

My point is that I am sending the jwt token, my business logic returns idUser to me and I want to download now, for example, all transactions where id_user = idUser

**Comment by henrymunro** [comment]
Can you use a relationship? 
https://hasura.io/docs/latest/actions/action-relationships/

E.g. in the Hassura UI for you action you can define and Array relationship from your action result to `streamId` From `idUser` to `id`

Then your query would look something like: 
```gql
query MyQuery {
  getIdUserNameAction(jwt: "...") {
    idUser
    streamId {
      id
      lastUpdated
     }
  }
}
```

**Comment by lukasz2161** [comment]
I can use relationship but then, for example, I will not be able to do limit, offset, where etc. in the query.

**Reply by talgat-ruby** [reply]
Then do two requests, or create new action which will do two requests.

**Reply by lukasz2161** [reply]
Could you show me an example of how to use the retrieved data for use in the next query?

**Reply by talgat-ruby** [reply]
1. Create action (sync I guess)
2. Make a request to `getIdUserNameAction`
3. Wait for a response and send it as argument(variable) to `steamId`

> Could you show me an example of how to use the retrieved data for use in the next query?

It depends on what language & framework you are using. But the above description should be sufficient to get the gist.
